### ISSUE TITLE
Does it work with SQL? 

### ISSUE THREAD

**Issue by Hexilee** [csmith-project/creduce#219]
Thx for your awesome project.

Recently I found some SQL queries cause logic bugs of DBMS, I would like to find which parts of it contribute to this bugs. 

Could C-Reduce do this job? How should I do it?

**Comment by eeide** [comment]
C-Reduce could possibly help—people have used it to reduce test cases written in various non-C/C++ languages including Java, JavaScript, Julia, Rust, and SMT-LIB.

I don't know of anybody who has used C-Reduce on SQL, though.

In terms of "how should you do it," are you asking for basic instructions (e.g., how does C-Reduce work?) or advanced tips?  I'm not sure that I have any "advanced tips" to offer, other than "you probably will want to disable the passes that are specific to C/C++."

If you send your question to the C-Reduce development mailing list, people there might have more general advice for you. See http://www.flux.utah.edu/mailman/listinfo/creduce-dev

Reducing SQL tes cases with C-Reduce sounds like a fun experiment!

**Comment by regehr** [comment]
I'll add that I'd love to hear either positive or negative results here!

**Comment by steakhal** [comment]
From my experience, `creduce` is a really versatile tool.
I'm using this on a daily basis reducing a wide variety of files.
Eg. I debugged Clang Cross Translation analysis crashes part of the Clang Static Analyzer which requires me to reduce some source files and text files as well.
For example `externalDefMap.txt` contains the information in lines, so I disabled all default passes and enabled only the `pass_lines` one. It worked out nicely. It would have taken hours me to reduce the mapping file by hand!

I can honestly recommend giving it a shot @Hexilee.
